src/Microsoft.Azure.Batch.SoftwareEntitlement.Server/Controllers/SoftwareEntitlementsController.cs

```diff
@@ -31,11 +32,16 @@ public class SoftwareEntitlementsController : Controller
         /// <param name="serverOptions">Options to use when handling requests.</param>
         /// <param name="logger">Reference to our logger for diagnostics.</param>
         /// <param name="lifetime">Lifetime instance to allow us to automatically shut down if requested.</param>
-        public SoftwareEntitlementsController(ServerOptions serverOptions, ILogger logger, IApplicationLifetime lifetime)
+        public SoftwareEntitlementsController(
+            ServerOptions serverOptions,
+            ILogger logger,
+            IApplicationLifetime lifetime,
+            EntitlementStore entitlementStore)
         {
             _serverOptions = serverOptions;
             _logger = logger;
             _lifetime = lifetime;
+            _entitlementStore = entitlementStore;
```

Comment:
The entitlement store needs to be part of the constructor of the `TokenVerifier` - the controller is responsible for gathering all the information and passing it to the verifier for a decision, nothing more (though it can complain if things are missing).

Once we get to the point of the call `_verifier.Verify()`, we shouldn't have anything else to do.

Reply:
At the moment, the `EntitlementStore` is longer-lived than the `TokenVerifier`. We _could_ make the token verifier long-lived, and pull the entitlement store into it, but I'm thinking it'd be cleaner to try and make a more obvious split between (1) extracting claims from the token (with minimal validation of the token and standard JWT claims) into a `NodeEntitlements` object, and (2) validating the request (application, cores, etc.) against the `NodeEntitlements` object.

If there are two separate components performing those two tasks, they end up with different dependencies, lifetimes and inputs, and I think it's easier to reason about what they're doing.